fontforge/python.c

```diff
@@ -8605,44 +8649,130 @@ static PyObject *PyFFGlyph_import(PyObject *self, PyObject *args,
     } else
 	ipp = &ip;
 
-    pt = strrchr(locfilename,'.');
-    if ( pt==NULL ) pt=locfilename;
+    if ( filetype==NULL ){
+        if ( locfilename==NULL ){
+            PyErr_SetString(PyExc_TypeError, "Expected a filetype or filename");
+            return NULL;
+        }
+        filetype = strrchr(locfilename, '.');
+        if ( filetype==NULL )
+        filetype=locfilename;
+        else
+        filetype += 1;
 
-    if ( strcasecmp(pt,".eps")==0 || strcasecmp(pt,".ps")==0 || strcasecmp(pt,".art")==0 ) {
-	SCImportPS(sc,((PyFF_Glyph *) self)->layer,locfilename,false,ipp);
+        char *lastcharacter = locfilename + strlen(locfilename) - 1;
+        if ( filetype >= lastcharacter)
+        return NULL;
     }
-    else if ( strcasecmp(pt,".svg")==0 ) {
-	SCImportSVG(sc,((PyFF_Glyph *) self)->layer,locfilename,NULL,0,false,ipp);
+
+
+    if ( strcasecmp(filetype,"eps")==0 || strcasecmp(filetype,"ps")==0 || strcasecmp(filetype,"art")==0 ) {
+        if ( locfilename==NULL ){
+            FILE *psfile = NULL;
+            #ifdef __unix__
```

Comment:
Please extract this ifdef...else...endif thing into a separate function and place it in, maybe, `fsys.c`.

Reply:
Done